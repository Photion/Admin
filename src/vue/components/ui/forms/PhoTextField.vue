<template>
  <div class="pho-field-wrapper">
    <label v-if="label" :for="id.unique" class="pho-label">
      {{ label }}
    </label>
    <input
      class="pho-field"
      :class="styles"
      :id="id.unique"
      :cy="id.family"
      :name="name"
      :type="type"
      :value="modelValue"
      :disabled="disabled"
      v-bind="$attrs"
      @input="onInput" />
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue';

import { useId, componentProps } from '~/src/vue/components/shared';
import { getFormProps } from '~/src/vue/components/ui/forms/forms';

export default defineComponent({

  props: {
    ...componentProps,
    ...getFormProps(String),
    type: {
      type: String,
      default: 'text',
    },
  },

  setup(props, context) {
    const onInput = ($event: InputEvent) => {
      const target = $event.target as HTMLInputElement;

      return context.emit('update:modelValue', target.value);
    };

    const styles = {
      disabled: props.disabled,
    };

    return {
      id: useId('field', props),
      styles,
      onInput,
    };
  },

});
</script>

<style scoped>
.disabled {
  color: gray;
}
</style>
