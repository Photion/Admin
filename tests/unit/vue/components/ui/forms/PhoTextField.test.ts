import { mount } from '@vue/test-utils';

import PhoTextField from '~/src/vue/components/ui/forms/PhoTextField.vue';

const getComponent = <T>(props: T) => {
  return mount(PhoTextField, {
    props: {
      name: 'pho-text-field-comp',
      uuid: '65441ec3-a9fa-4e8c-b1e5-79d7d001e634',
      modelValue: 'abc',
      ...props,
    },
  });
};

describe('unit.vue.components.ui.forms.PhoTextField', () => {

  it('Mounts', () => {
    const wrapper = getComponent({});
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Sets the attributes', () => {
    const wrapper = getComponent({});
    const attributes = wrapper.get('[cy]').attributes();

    expect(attributes).toMatchSnapshot();
  });

  it('Renders the label', async () => {
    const wrapper = getComponent({ label: 'The Best Fruit is...', modelValue: 'Blueberry' });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Renders a disabled field', () => {
    const wrapper = getComponent({ disabled: true });
    expect(wrapper.element).toMatchSnapshot();
  });

  ['text', 'password', 'number', 'date'].forEach((type) => {
    it(`Renders a '${type}' field`, () => {
      const wrapper = getComponent({ type });
      expect(wrapper.element).toMatchSnapshot();
    });

  });

});
