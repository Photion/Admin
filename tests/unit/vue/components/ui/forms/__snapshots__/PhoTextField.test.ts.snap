// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit.vue.components.ui.forms.PhoTextField Mounts 1`] = `
<div
  class="pho-field-wrapper"
>
  <!--v-if-->
  <input
    class="pho-field"
    cy="field:pho-text-field-comp"
    id="field:pho-text-field-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
    name="pho-text-field-comp"
    type="text"
  />
</div>
`;

exports[`unit.vue.components.ui.forms.PhoTextField Renders a 'date' field 1`] = `
<div
  class="pho-field-wrapper"
>
  <!--v-if-->
  <input
    class="pho-field"
    cy="field:pho-text-field-comp"
    id="field:pho-text-field-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
    name="pho-text-field-comp"
    type="date"
  />
</div>
`;

exports[`unit.vue.components.ui.forms.PhoTextField Renders a 'number' field 1`] = `
<div
  class="pho-field-wrapper"
>
  <!--v-if-->
  <input
    class="pho-field"
    cy="field:pho-text-field-comp"
    id="field:pho-text-field-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
    name="pho-text-field-comp"
    type="number"
  />
</div>
`;

exports[`unit.vue.components.ui.forms.PhoTextField Renders a 'password' field 1`] = `
<div
  class="pho-field-wrapper"
>
  <!--v-if-->
  <input
    class="pho-field"
    cy="field:pho-text-field-comp"
    id="field:pho-text-field-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
    name="pho-text-field-comp"
    type="password"
  />
</div>
`;

exports[`unit.vue.components.ui.forms.PhoTextField Renders a 'text' field 1`] = `
<div
  class="pho-field-wrapper"
>
  <!--v-if-->
  <input
    class="pho-field"
    cy="field:pho-text-field-comp"
    id="field:pho-text-field-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
    name="pho-text-field-comp"
    type="text"
  />
</div>
`;

exports[`unit.vue.components.ui.forms.PhoTextField Renders a disabled field 1`] = `
<div
  class="pho-field-wrapper"
>
  <!--v-if-->
  <input
    class="pho-field disabled"
    cy="field:pho-text-field-comp"
    disabled=""
    id="field:pho-text-field-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
    name="pho-text-field-comp"
    type="text"
  />
</div>
`;

exports[`unit.vue.components.ui.forms.PhoTextField Renders the extra attr over the input 1`] = `
<div
  class="pho-field-wrapper"
  min="1900"
>
  <!--v-if-->
  <input
    class="pho-field"
    cy="field:pho-text-field-comp"
    id="field:pho-text-field-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
    min="1900"
    name="pho-text-field-comp"
    type="text"
  />
</div>
`;

exports[`unit.vue.components.ui.forms.PhoTextField Renders the extra attr over the input 2`] = `
<div
  class="pho-field-wrapper"
  max="2100"
>
  <!--v-if-->
  <input
    class="pho-field"
    cy="field:pho-text-field-comp"
    id="field:pho-text-field-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
    max="2100"
    name="pho-text-field-comp"
    type="text"
  />
</div>
`;

exports[`unit.vue.components.ui.forms.PhoTextField Renders the extra attr over the input 3`] = `
<div
  class="pho-field-wrapper"
  max="2100"
  min="1900"
>
  <!--v-if-->
  <input
    class="pho-field"
    cy="field:pho-text-field-comp"
    id="field:pho-text-field-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
    max="2100"
    min="1900"
    name="pho-text-field-comp"
    type="text"
  />
</div>
`;

exports[`unit.vue.components.ui.forms.PhoTextField Renders the label 1`] = `
<div
  class="pho-field-wrapper"
>
  <label
    class="pho-label"
    for="field:pho-text-field-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
  >
    The Best Fruit is...
  </label>
  <input
    class="pho-field"
    cy="field:pho-text-field-comp"
    id="field:pho-text-field-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634"
    name="pho-text-field-comp"
    type="text"
  />
</div>
`;

exports[`unit.vue.components.ui.forms.PhoTextField Sets the attributes 1`] = `
Object {
  "class": "pho-field",
  "cy": "field:pho-text-field-comp",
  "id": "field:pho-text-field-comp:65441ec3-a9fa-4e8c-b1e5-79d7d001e634",
  "name": "pho-text-field-comp",
  "type": "text",
}
`;
